Microsoft.Azure.Cosmos/src/FeedIteratorCore.cs

```
@@ -110,17 +201,36 @@ public override bool TryGetContinuationToken(out string continuationToken)
             return true;
         }
 
-        internal static string GetContinuationToken(ResponseMessage httpResponseMessage)
+        private async Task<TryCatch<FeedTokenInternal>> TryInitializeFeedTokenAsync(CancellationToken cancellationToken)
         {
-            return httpResponseMessage.Headers.ContinuationToken;
-        }
+            string containerRId = string.Empty;
+            if (this.containerCore != null)
+            {
+                try
+                {
+                    containerRId = await this.containerCore.GetRIDAsync(cancellationToken);
+                }
+                catch (CosmosException cosmosException)
```

Why not catch all Exceptions?

---

Good point, done